frontend/src/containers/Table.js

```diff
@@ -0,0 +1,268 @@
+import update from 'immutability-helper';
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+
+import { getZoomIntoWindow, deleteRequest } from '../api';
+import { getTableId, getTable } from '../reducers/tables';
+
+import {
+  updateTableSelection,
+  deselectTableItems,
+  collapseTableRow,
+} from '../actions/TableActions';
+import { showIncludedView } from '../actions/ViewActions';
+import { openModal } from '../actions/WindowActions';
+
+import { containerPropTypes } from '../utils/tableHelpers';
+import { mapIncluded } from '../utils/documentListHelper';
+
+import Table from '../components/table/Table';
+
+class TableContainer extends PureComponent {
+  componentWillUnmount() {
+    const {
+      showIncludedView,
+      viewId,
+      windowId,
+      isIncluded,
+      isModal,
+    } = this.props;
+
+    if (!isIncluded) {
+      const identifier = isModal ? viewId : windowId;
+
+      showIncludedView({
+        id: identifier,
+        showIncludedView: false,
+        windowId,
+        viewId,
+      });
+    }
+  }
+
+  getAllLeaves = () => {
+    const { rows, selected } = this.props;
+    let leafs = [];
+    let leafsIds = [];
+
+    rows.map((item) => {
+      if (item.id === selected[0]) {
+        leafs = mapIncluded(item);
+      }
+    });
+
+    leafs.map((item) => {
+      leafsIds = leafsIds.concat(item.id);
+    });
+
+    this.handleSelect(leafsIds);
+  };
+
+  handleSelect = (ids, cb) => {
+    const {
+      updateTableSelection,
+      windowId,
+      viewId,
+      docId,
+      tabId,
+      keyProperty,
+    } = this.props;
+    let newSelected = [];
```

Comment:
can we move all this logic from L:70-78 to a separate function that can live outside the Table component  in tableHelpers ?  let newSelection = getSelectionForIds(ids). Goal is to have the Table as thin as possible

Reply:
But this is in the container, which is ~270 LOC's so I don't really see much sense. Also we'd then have to pass like 6 or 7 parameters.